src/libraries/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.cs

```diff
@@ -158,14 +158,25 @@ public override object Invoke(BindingFlags invokeAttr, Binder? binder, object?[]
                 throw new TargetParameterCountException(SR.Arg_ParmCnt);
             }
 
+            // Array construction is a special case.
+            if (DeclaringType is not null && DeclaringType.IsArray)
+            {
+                Span<int> args = stackalloc int[actualCount];
```

Comment:
Is adding `stackalloc` here going to visibly regress the non-array path through the method?

Reply:
At this point it is simply a call out to a dedicated FCall and I observe no impact.